### PR TITLE
947 disable prefix list hierarchy

### PR DESCRIPTION
**Pull request by lampwins**
### Fixes: #947 

This PR introduces a new configuration setting `DISABLE_PREFIX_LIST_HIERARCHY` which defaults to `False`. When enabled, this setting manipulates the `nautobot.ipam.views.PrefixListView`'s queryset such that it does not use `annotate_tree()` to retrieve parent/child relationships, which proved to be the performance bottleneck.

### CODE REVIEW

File: nautobot/ipam/tests/integration/test_prefixes.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        status = Status.objects.get(slug="active")
+        Prefix(prefix=netaddr.IPNetwork("10.0.0.0/16"), status=status).validated_save()
+        Prefix(prefix=netaddr.IPNetwork("10.0.0.0/24"), status=status).validated_save()
+
+        # Navigate to Prefix list view
+        self.browser.visit(self.live_server_url)
+        self.browser.links.find_by_partial_text("IPAM").click()
+        self.browser.links.find_by_partial_text("Prefixes").click()
+
+        self.assertTrue(self.browser.find_by_tag("tr")[2].find_by_tag("i").first.has_class("mdi-circle-small"))
```
Comment (glennmatthews) [inline nautobot/ipam/tests/integration/test_prefixes.py:41]: This feels fragile and a bit non-obvious to me. Can we at least assert that the `tr` that we find has the correct "10.0.0.0/24" string in it as well?

File: nautobot/ipam/tests/integration/test_prefixes.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        status = Status.objects.get(slug="active")
+        Prefix(prefix=netaddr.IPNetwork("10.0.0.0/16"), status=status).validated_save()
+        Prefix(prefix=netaddr.IPNetwork("10.0.0.0/24"), status=status).validated_save()
+
+        # Navigate to Prefix list view
+        self.browser.visit(self.live_server_url)
+        self.browser.links.find_by_partial_text("IPAM").click()
+        self.browser.links.find_by_partial_text("Prefixes").click()
+
+        self.assertEqual(len(self.browser.find_by_tag("tr")[2].find_by_tag("i")), 0)  # no such element
```
Comment (glennmatthews) [inline nautobot/ipam/tests/integration/test_prefixes.py:57]: Just to confirm, in this case we *should* find the "10.0.0.0/24" `tr` still, it just shouldn't have the indentation/bullet?

File: nautobot/core/settings.py [hunk 3]
```diff
@@ -657,3 +657,8 @@
 
 # Undocumented link in the bottom right of the footer which is meant to persist any custom branding changes.
 BRANDING_POWERED_BY_URL = "https://nautobot.readthedocs.io/"
+
+# Do not annotate the depth of prefix parent/child relationships in the prefix list view
+# TODO: we should consider this a temporary workaround available to users
+# for https://github.com/nautobot/nautobot/issues/947, until a proper data model change is made.
+DISABLE_PREFIX_LIST_HIERARCHY = is_truthy(os.getenv("NAUTOBOT_DISABLE_PREFIX_LIST_HIERARCHY", False))
```
Comment (glennmatthews) [inline nautobot/core/settings.py:664]: Would it be useful/desirable to make this an admin-UI-configurable setting under Constance?

File: nautobot/core/settings.py [hunk 4]
```diff
@@ -445,6 +445,10 @@
         90,
         "Number of days to retain object changelog history.\nSet this to 0 to retain changes indefinitely.",
     ],
+    "DISABLE_PREFIX_LIST_HIERARCHY": [
+        False,
+        "Dissable rendering parent/child relationships in the IPAM Prefix list view and instead " "show a flat list.",
```
Comment (glennmatthews) [inline nautobot/core/settings.py:450]:
```suggestion
        "Disable rendering parent/child relationships in the IPAM Prefix list view and instead show a flat list.",
```

**Review by glennmatthews — APPROVED** [review]

File: nautobot/core/settings.py (hunk 4 above)
Comment (lampwins) [reply nautobot/core/settings.py:450]: I will never get this right :)

**Pull request merged**